R/cube-dims.R

```diff
@@ -199,3 +200,46 @@ is.selectedArrayDim <- function (dim) {
 
     return(FALSE)
 }
+
+
+#' Get dimension type
+#'
+#' This function returns the specific type of each cube dimension. This is useful
+#' when cubes contain categorical array or multiple response variables because it
+#' identifies the dimensions of the cube which refer to the differ parts of
+#' array variable:
+#' - `ca_items`: Categorical array items
+#' - `ca_categories`: The categories of the categorical array
+#' - `mr_items`: Multiple response options or items
+#' - `mr_selections`: The selection status for a multiple response variable
+#'
+#' @return A character vector. This is identical to `types()` except that
+#' the array variable types are more specific.
+#' @param x a CrunchCube or CubeDims object
+#'
+#' @return a character vector of dimension types
+#' @export
+#' @keywords internal
+getDimType <-  function (x) {
+    if (inherits(x, "CrunchCube")) {
+        x <- x@dims
+    }
+    vars <- variables(x)
+    out <- types(vars)
+    out[is.selectedDimension(x)] <- "mr_selections"
+    for (i in seq_along(vars)) {
+        if (i == length(vars)) {
+            break
+        }
+        if (out[i + 1] == "mr_selections") {
+            out[i] <- "mr_items"
+            next
+        }
+        if (out[i] == "subvariable_items") {
+            out[i] <- "ca_items"
+            out[i + 1] <- "ca_categories"
+        }
+    }
+    names(out) <- names(vars)
+    return(out)
+}
```

Comment:
This is nice, I wonder if it would be possible to unify this code with the code like https://github.com/Crunch-io/rcrunch/blob/master/R/cube-dims.R#L39-L41 and https://github.com/Crunch-io/rcrunch/blob/master/R/cube-dims.R#L167-L192 . It would be good, if possible, to handle all of this inspecting and labelling in one place.